src/solvers/smt2/smt2_conv.cpp

```diff
@@ -3122,6 +3122,31 @@ void smt2_convt::convert_relation(const binary_relation_exprt &expr)
     convert_expr(expr.op1());
     out << ")";
   }
+  else if(op_type.id() == ID_pointer)
+  {
+    const exprt same_object = ::same_object(expr.op0(), expr.op1());
+
+    out << "(and ";
+    convert_expr(same_object);
+
+    out << " (";
+    if(expr.id() == ID_le)
+      out << "bvule";
+    else if(expr.id() == ID_lt)
+      out << "bvult";
+    else if(expr.id() == ID_ge)
+      out << "bvuge";
+    else if(expr.id() == ID_gt)
+      out << "bvugt";
+
```

Comment:
You are doing a signed comparison in bv_pointers but an unsigned one here?

Reply:
Thank you, this is now fixed (copy&paste mistake as this was previously handled in the same case as `ID_unsigned`).